tests/resolution/test_timeout_sync.py

```
@@ -176,6 +177,38 @@ async def make_chain(depth: int) -> Any:
     assert await promise == "done"
 
 
+async def test_sync_timeout_spans_multiple_levels_elapsed_time() -> None:
+    """The total elapsed time until TimeoutError should be roughly
+    equal to the timeout value — the deadline must carry across
+    all unpacking levels, not restart per level."""
+
+    async def make_chain(depth: int) -> Any:
```

Ruff (0.15.7)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `make_chain`

(ANN401)